crates/shim-kvm/src/spin.rs

```diff
@@ -85,3 +86,26 @@ impl<A> RwLocked<A> {
         self.inner.write()
     }
 }
+
+/// A wrapper around spin::Lazy to permit trait implementations.
```

Comment:
I don't want to implement `Deref<Target = T>` for every struct, which has an `inner: Lazy<T>`

Reply:
> I don't want to implement `Deref<Target = T>` for every struct, which has an `inner: Lazy<T>`

Chances are, you won't have to. In this specific case, you could define all methods on `LockedEnarxAllocator`, so there'd be no reason to access the inner type.

Besides that, implementing a trait on `Lazy` really isn't a common thing though: The global allocator is special because because it's a language item that needs `GlobalAlloc` implemented directly on the type of the static and can't use `Lazy`'s `Deref` impl. In most if not all other cases, you don't need to implement on a trait on `Lazy<T>` (or `Lazied`), but can instead implement it on `T` and then rely on `Lazy`'s `Deref` impl when using the value.

Reply:
ok, you convinced me :) changed